Core/include/Acts/Propagator/detail/actor_list_implementation.hpp

```diff
@@ -0,0 +1,104 @@
+  {
+    if constexpr (ActorHasActWithoutResult<actor_t, propagator_state_t,
+                                           stepper_t, navigator_t, Args...>) {
+      actor.act(state, stepper, navigator, std::forward<Args>(args)...);
+      return;
+    }
+
+    if constexpr (ActorHasActWithResult<actor_t, propagator_state_t, stepper_t,
+                                        navigator_t, Args...>) {
+      actor.act(state, stepper, navigator,
+                state.template get<detail::result_type_t<actor_t>>(),
+                std::forward<Args>(args)...);
+      return;
+    }
+  }
+
+  template <typename actor_t, typename propagator_state_t, typename stepper_t,
+            typename navigator_t, typename... Args>
+  static bool checkAbort(const actor_t& actor, propagator_state_t& state,
+                         const stepper_t& stepper, const navigator_t& navigator,
+                         Args&&... args)
+    requires(
+        Actor<actor_t, propagator_state_t, stepper_t, navigator_t, Args...>)
+  {
+    if constexpr (ActorHasAbortWithoutResult<actor_t, propagator_state_t,
+                                             stepper_t, navigator_t, Args...>) {
+      return actor.checkAbort(state, stepper, navigator,
+                              std::forward<Args>(args)...);
+    }
+
+    if constexpr (ActorHasAbortWithResult<actor_t, propagator_state_t,
+                                          stepper_t, navigator_t, Args...>) {
+      return actor.checkAbort(
+          state, stepper, navigator,
+          state.template get<detail::result_type_t<actor_t>>(),
+          std::forward<Args>(args)...);
+    }
+
+    return false;
+  }
+};
+
+}  // namespace
+
+template <typename... actors_t>
+struct actor_list_impl {
+  template <typename propagator_state_t, typename stepper_t,
+            typename navigator_t, typename... Args>
+  static void act(const std::tuple<actors_t...>& actor_tuple,
+                  propagator_state_t& state, const stepper_t& stepper,
+                  const navigator_t& navigator, Args&&... args) {
+    std::apply(
+        [&](const actors_t&... actor) {
+          (actor_caller::act(actor, state, stepper, navigator,
+                             std::forward<Args>(args)...),
+           ...);
+        },
+        actor_tuple);
+  }
+
+  template <typename propagator_state_t, typename stepper_t,
+            typename navigator_t, typename... Args>
+  static bool checkAbort(const std::tuple<actors_t...>& actor_tuple,
+                         propagator_state_t& state, const stepper_t& stepper,
+                         const navigator_t& navigator, Args&&... args) {
+    return std::apply(
+        [&](const actors_t&... actor) {
+          return (actor_caller::checkAbort(actor, state, stepper, navigator,
+                                           std::forward<Args>(args)...) ||
+                  ...);
+        },
+        actor_tuple);
```

Comment:
I think (might be wrong) previously, we would not call subsequent aborters when the first one indicated an abort condition. This I don't think does that anymore. Do we want that? Do we care?

Reply:
I missed the `||` in the fold expression. You're right.